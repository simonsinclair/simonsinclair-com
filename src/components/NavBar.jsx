import PropTypes from 'prop-types';
import { withRouter } from 'next/router';
import Link from 'next/link';

const Nav = ({ router }) => {
  const { route } = router;

  return (
    <nav className="nav-bar">
      <ul className="nav-list">
        <li>
          <Link href="/">
            <a className={route === '/' ? 'active' : ''}>Home</a>
          </Link>
        </li>
      </ul>

      <style jsx>
        {`
          .nav-bar {}
          .active {
            border-bottom: 2px solid;
          }
        `}
      </style>
    </nav>
  );
};

Nav.propTypes = {
  router: PropTypes.shape({}).isRequired,
};

export default withRouter(Nav);
