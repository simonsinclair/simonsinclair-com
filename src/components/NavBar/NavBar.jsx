import PropTypes from 'prop-types';
import { withRouter } from 'next/router';
import Link from 'next/link';

const Nav = ({ router }) => {
  const { route } = router;

  return (
    <nav className="nav-bar">
      <ul className="nav-list nav-list--inline">
        <li>
          <Link href="/">
            <a className={route === '/' ? 'active' : ''}>Home</a>
          </Link>
        </li>
      </ul>

      <style jsx>
        {`
          .nav-bar {}
          .active {}
        `}
      </style>
    </nav>
  );
};

Nav.propTypes = {
  router: PropTypes.shape({}).isRequired,
};

export default withRouter(Nav);
