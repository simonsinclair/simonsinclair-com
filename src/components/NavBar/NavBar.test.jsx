import { render } from '@testing-library/react';

import NavBar from './NavBar';

describe('<NavBar /> is a component that', () => {
  it('renders and matches its snapshot', () => {
    const { container } = render(<NavBar />);
    expect(container).toMatchSnapshot();
  });
});
