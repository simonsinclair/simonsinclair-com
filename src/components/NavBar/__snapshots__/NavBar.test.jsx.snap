// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> is a component that renders and matches its snapshot 1`] = `
<div>
  <nav
    class="nav-bar"
  >
    <ul
      class="nav-list nav-list--inline"
    >
      <li>
        <a
          class=""
          href="/"
        >
          Home
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
